pkg/runtime/codec.go

```diff
@@ -80,12 +80,14 @@ func EncodeOrDie(e Encoder, obj Object) string {
 // invokes the ObjectCreator to instantiate a new gvk. Returns an error if the typer cannot find the object.
 func UseOrCreateObject(t ObjectTyper, c ObjectCreater, gvk unversioned.GroupVersionKind, obj Object) (Object, error) {
 	if obj != nil {
-		into, _, err := t.ObjectKinds(obj)
+		kinds, _, err := t.ObjectKinds(obj)
 		if err != nil {
 			return nil, err
 		}
-		if gvk == into[0] {
```

Comment:
This was adapted from previous code that did not handle multiple into.  So it's correct to iterate over kinds here (now that the refactor to t.ObjectKinds() from t.ObjectKind() has been completed).